src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/event.py

```diff
@@ -751,14 +762,14 @@ def carp_backup(self, fobj, ifname, vhid, event, user_override):
                     if ret and ret[0]['srv_enable']:
                         self.run_call(f'service.{verb}', i, {'ha_propagate': False})
 
-                run('LD_LIBRARY_PATH=/usr/local/lib /usr/local/sbin/enc_helper detachall')
-
-                if fobj['phrasedvolumes']:
-                    self.logger.warn('Setting passphrase from master')
-                    run('LD_LIBRARY_PATH=/usr/local/lib /usr/local/sbin/enc_helper syncfrompeer')
+                detach_all_job = self.middleware.call_sync('failover.encryption_detachall')
+                detach_all_job.wait_sync()
+                if detach_all_job.error:
+                    self.logger.error('Failed to detach geli providers: %s', detach_all_job.error)
 
+                self.middleware.call_sync('failover.call_remote', ['failover.sync_keys_with_remote_node'])
```

Comment:
This seems wrong?

self.middleware.call_sync('failover.call_remote', 'failover.sync_keys_with_remote_node')

Reply:
It is wrong, thank you for catching that. I would've picked it up after i rolled a new build : p